checks.d/process.py

```diff
@@ -42,9 +49,11 @@ def find_pids(self, search_string, exact_match, ignore_denied_access):
                     except psutil.NoSuchProcess:
                         self.log.warning('Process disappeared while scanning')
                     except psutil.AccessDenied, e:
-                        self.log.error('Access denied to %s process' % string)
-                        self.log.error('Error: %s' % e)
-                        if not ignore_denied_access:
+                        if not ignore_denied_access and proc.pid not in catched_access_denied:
+                            self.log.error('Access denied to %s process' % string)
+                            self.log.error('Error: %s' % e)
+                            # allows to raise only once for each failing process
+                            self.catched_access_denied.append(proc.pid)
```

Comment:
You always append to this list but never delete anything.
If the processes watched are rotating in a way it could be bad. You should be able to identify a unique process by using its creation time I guess?